proparse/src/main/java/org/prorefactor/proparse/antlr4/PostLexer.java

```diff
@@ -0,0 +1,305 @@
+    // Preserve the current position for listing, before evaluating the expression.
+    // We can't just write to listing here, because the expression evaluation may
+    // find macro references to list.
+    int currLine = currToken.getLine();
+    int currCol = currToken.getCharPositionInLine();
+    boolean evaluate = true;
+    // Don't evaluate if we're consuming from an outer &IF
+    if (prepro.getConsuming() - 1 > 0)
+      evaluate = false;
+    // Don't evaluate if we're already done with this &IF
+    if (preproIfVec.getLast().done)
+      evaluate = false;
+    boolean isTrue = preproIfCond(evaluate);
+    prepro.getLstListener().preproElseIf(currLine, currCol);
+    PreproIfState preproIfState = preproIfVec.getLast();
+    if (isTrue && (!preproIfState.done)) {
+      preproIfState.done = true;
+      preproIfState.consuming = false;
+    } else {
+      if (!preproIfState.consuming) {
+        preproIfState.consuming = true;
+        preproconsume();
+      }
+      // else: already consuming. no change.
+    }
+  }
+
+  private void preproEndif() throws IOException {
+    LOGGER.trace("Entering preproEndif()");
+    prepro.getLstListener().preproEndIf(currToken.getLine(), currToken.getCharPositionInLine());
+    // XXX Got a case where removeLast() fails with NoSuchElementException
+    if (!preproIfVec.isEmpty())
+      preproIfVec.removeLast();
+  }
+
+  private boolean preproIfCond(boolean evaluate) throws IOException, TokenStreamException, RecognitionException {
+    LOGGER.trace("Entering preproIfCond()");
+
+    
+    // Notes
+    // An &IF here in this &IF condition is not legal. Progress would barf on it.
+    // That allows us to simply use a global flag to watch for &THEN.
+
+    List<ProToken> tokenVector = new ArrayList<>();
+    boolean done = false;
+    while (!done) {
+      getNextToken();
+      switch (currToken.getType()) {
+        case PreprocessorParser.EOF:
+          throwMessage("Unexpected end of input after &IF or &ELSEIF");
+          break;
+        case PreprocessorParser.AMPTHEN:
+          done = true;
+          break;
+        case PreprocessorParser.DEFINED:
+          if (evaluate)
+            // If not evaluating, just discard
+            tokenVector.add(defined());
+          break;
+        case PreprocessorParser.COMMENT:
+        case PreprocessorParser.WS:
+        case PreprocessorParser.PREPROCESSTOKEN:
+          break;
+        default:
+          if (evaluate)
+            // If not evaluating, just discard
+            tokenVector.add(currToken);
+      }
+    }
+
+    // If it's blank or the the evaluate argument is false, we don't evaluate
+    if (tokenVector.isEmpty() || !evaluate)
+      return false;
+    else {
+      CommonTokenStream cts = new CommonTokenStream(new ListTokenSource(tokenVector));
+      PreprocessorParser parser = new PreprocessorParser(cts);
+      PreproEval eval = new PreproEval();
+      Boolean b = eval.visitPreproIfEval(parser.preproIfEval());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "b". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1488)